osu.Game/Collections/CollectionFilterMenuItem.cs

```diff
@@ -61,6 +56,10 @@ public AllBeatmapsCollectionFilterMenuItem()
             : base("All beatmaps")
         {
         }
+
+        public override bool Equals(CollectionFilterMenuItem? other) => other is AllBeatmapsCollectionFilterMenuItem;
+
+        public override int GetHashCode() => 1;
```

Comment:
scary but i'll assume there's some theory behind this.

Reply:
The way to think about it is as such: if two objects are equal, they must have the same hash code, but two objects with the same hash code are not necessarily equal.

So a constant hash code spec such as this ensures that all instances of the class have the same hash code, and since the equality member only checks type, this satisfies the first part.

There is no actual data to base the hash code implementation on. If anything it'll just be a little inefficient in a dictionary, but the assumption is that there's only ever going to be one of these per dropdown.